doc/common/Cadl-Project-Scripts.md

```diff
@@ -0,0 +1,127 @@
+The file should be checked into this location `./eng/emitter-package.json`
+
+Example
+
+```json
+{
+    "main": "dist/src/index.js",
+    "dependencies": {
+      "@azure-tools/cadl-csharp": "0.1.11-beta.20230123.1"
+    }
+}
+```
+
+Note that cadl compile currently requires the "main" line to be there.
+
+### Language-Settings.ps1
+
+There are three methods you can write in your language repo to adjust the behavior of the scripts one of which is required.
+For each of these replace `${Language}` with the language identifier in your repo.  If you don't know what this is you can look at `./eng/scripts/Language-Settings.ps1` in your language repo and you will find other functions that match this pattern that already exist.
+
+#### Get-${Language}-EmitterName *(Required)*
+
+This function simply returns the emitter name string.
+
+Example
+
+```powershell
+function Get-dotnet-EmitterName() {
+  return "@azure-tools/cadl-csharp"
+}
+```
+
+#### Get-${Language}-EmitterPackageJsonPath (Optional)
+
+This function allows you to specify the location and name of the emitter package.json to use.  If this is omitted the script will assume the default location listed above `./eng/emitter-package.json`.  The path must be absolute.
+
+Example
+
+```powershell
+function Get-dotnet-EmitterPackageJsonPath() {
+  return "D:\SomeOtherLocation\some-other-emitter-package.json"
+}
+```
+
+#### Get-${Language}-EmitterAdditionalOptions (Optional)
+
+This function allows you to append additional `--option` arguments that will be passed into cadl compile.  One example of this is the `emitter-output-dir`.  For dotnet we want the location of the generated files to be `{projectDir}/src` however in other languages they will have other conventions.  This method will take in a fully qualified path to the project directory so you can construct your relative path to that as the output.
+
+Example
+
+```powershell
+function Get-dotnet-EmitterAdditionalOptions([string]$projectDirectory) {
+  return "--option @azure-tools/cadl-csharp.emitter-output-dir=$projectDirectory/src"
+}
+```
+
+## Cadl-Project-Sync.ps1
+
+This is the first script that should be called and can be found at `./eng/common/scripts/Cadl-Project-Sync.ps1`.  It takes in one parameter which is the root directory of the project which is typically one layer lower than the service directory.  As an example for dotnet this is `./sdk/openai/Azure.AI.OpenAI` where `openai` is the service directory and `Azure.AI.OpenAI` is the project directory.
+
+This script will create a sparse check out at the root of your current repository named after the project directory, automatically filter to only the files in the directory defined in cadl-location.yaml, and sync to the sha defined in cadl-location.yaml.
```

Comment:
Sometimes a cadl project will reference a shared library under another shared project folder. In this case, sparse check out should happen on the service root folder rather than project folder.

Reply:
This is what additional directories are for, you can set that to anything in the spec repo without needing to always pull down the entire service directory.